src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExporter.cs

```diff
@@ -40,57 +41,70 @@ public class DynatraceMetricsExporter : MetricExporter
+
+        internal DynatraceMetricsExporter(DynatraceExporterOptions options, ILogger<DynatraceMetricsExporter> logger, HttpClient client)
         {
             this._options = options ?? new DynatraceExporterOptions();
             this._logger = logger ?? NullLogger<DynatraceMetricsExporter>.Instance;
-            logger.LogDebug("Dynatrace Metrics Url: {Url}", this._options.Url);
-            this._httpClient = new HttpClient();
-            if (!string.IsNullOrEmpty(options.ApiToken))
+            this._logger.LogDebug("Dynatrace Metrics Url: {Url}", this._options.Url);
+            this._httpClient = client;
+            if (!string.IsNullOrEmpty(this._options.ApiToken))
             {
                 this._httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Api-Token", this._options.ApiToken);
             }
-            var defaultLabels = new List<KeyValuePair<string, string>>();
-            if (options.DefaultDimensions != null) defaultLabels.AddRange(options.DefaultDimensions);
-            if (options.OneAgentMetadataEnrichment)
-            {
-                var enricher = new OneAgentMetadataEnricher(this._logger);
-                enricher.EnrichWithDynatraceMetadata(defaultLabels);
-            }
-            this._serializer = new DynatraceMetricSerializer(options.Prefix, defaultLabels);
+            this._httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(new ProductHeaderValue("opentelemetry-metric-dotnet")));
+            this._serializer = new DynatraceMetricSerializer(this._logger, this._options.Prefix, this._options.DefaultDimensions, this._options.EnrichWithDynatraceMetadata);
         }
 
         public override async Task<ExportResult> ExportAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken)
         {
-            var sw = Stopwatch.StartNew();
-            var httpRequest = new HttpRequestMessage(HttpMethod.Post, this._options.Url);
-            var sb = new StringBuilder();
-            foreach (var metric in metrics)
-            {
-                _serializer.SerializeMetric(sb, metric);
-            }
+            // split all metrics into batches of DynatraceMetricApiConstants.PayloadLinesLimit lines
+            var chunked = metrics
+            .Select((val, i) => new { val, batch = i / DynatraceMetricApiConstants.PayloadLinesLimit })
+            .GroupBy(x => x.batch)
+            .Select(x => x.Select(v => v.val));
+
+            var exportResults = new List<ExportResult>();
 
-            var metricLines = sb.ToString();
-            _logger.LogDebug(metricLines);
-            httpRequest.Content = new StringContent(metricLines);
-            try
+            foreach (var chunk in chunked)
             {
-                var response = await this._httpClient.SendAsync(httpRequest);
-                if (response.IsSuccessStatusCode)
+                var sw = Stopwatch.StartNew();
```

Comment:
I think we can drop this.

Reply:
It debug-logs the response time. I dropped it.